avm/res/logic/integration-account/main.bicep

```diff
@@ -0,0 +1,511 @@
+    params: {
+      name: certificate.name
+      location: location
+      integrationAccountName: integrationAccount.name
+      key: certificate.?key
+      metadata: certificate.?metadata
+      publicCertificate: certificate.?publicCertificate
+      tags: certificate.?tags ?? tags
+    }
+  }
+]
+
+module integrationAccount_agreements 'agreement/main.bicep' = [
+  for (agreement, index) in (agreements ?? []): {
+    name: '${uniqueString(deployment().name, location)}-integrationAccount-Agreement-${index}'
+    params: {
+      name: agreement.name
+      location: location
+      integrationAccountName: integrationAccount.name
+      agreementType: agreement.agreementType
+      guestIdentity: agreement.guestIdentity
+      guestPartner: agreement.guestPartner
+      hostIdentity: agreement.hostIdentity
+      hostPartner: agreement.hostPartner
+      content: agreement.content
+      metadata: agreement.?metadata
+      tags: agreement.?tags ?? tags
+    }
+    dependsOn: [
+      integrationAccount_partners
+    ]
+  }
+]
+
+resource integrationAccount_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(
+      integrationAccount.id,
+      roleAssignment.principalId,
+      roleAssignment.roleDefinitionId
+    )
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: integrationAccount
+  }
+]
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The resource ID of the integration account.')
+output resourceId string = integrationAccount.id
+
+@description('The name of the integration account.')
+output name string = integrationAccount.name
+
+@description('The location the resource was deployed into.')
+output location string = integrationAccount.location
+
+@description('The resource group the integration account was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+// ================ //
+// Definitions      //
+// ================ //
+
+import { b2bPartnerContentType } from 'partner/main.bicep'
+@description('The type for an integration account partner.')
+type integrationAccountPartnerType = {
+  @description('Required. The Name of the partner resource.')
+  name: string
+
+  @description('Optional. The B2B partner content settings.')
+  b2bPartnerContent: b2bPartnerContentType?
+
+  @description('Optional. The partner metadata.')
+  metadata: object?
```

Comment:
This and other metadata references should be replaced with the resource derived type I commented in the individual child-module files

Reply:
Hey @lsnoddy,
thanks for the comment. That is ... something. I'll try to reproduce this and find a solution

Reply:
Alright, @lsnoddy, I was able to replicate the issue myself. The best solution I was able to come up with which seems to work
- Use Resource-Dervied-Types for all 'metadata' parameters but for the modules
  - partners
  - schemas
  - maps

For the parameter, use instead
```bicep
@description('Optional. The metadata.')
param metadata {
  @description('Optional. A metadata key-value pair.')
  *: string?
}? // Resource-derived type fails PSRule
```

for their types in the parent module 'integration-account', instead use
```bicep
  @description('Optional. The metadata.')
  metadata: {
    @description('Optional. A metadata key-value pair.')
    *: string?
  }? // Resource-derived type fails PSRule
```